xds/internal/balancer/edsbalancer/xds_client_wrapper.go

```diff
@@ -48,44 +48,75 @@ var (
 )
 
 type loadStoreWrapper struct {
-	mu         sync.RWMutex
+	mu      sync.RWMutex
+	service string
+	// Both store and perCluster will be nil if load reporting is disabled (EDS
+	// response doesn't have LRS server name).
 	store      *load.Store
-	service    string
 	perCluster load.PerClusterReporter
 }
 
-func (lsw *loadStoreWrapper) update(store *load.Store, service string) {
+func (lsw *loadStoreWrapper) updateServiceName(service string) {
 	lsw.mu.Lock()
 	defer lsw.mu.Unlock()
-	if store == lsw.store && service == lsw.service {
+	if lsw.service == service {
 		return
 	}
-	lsw.store = store
 	lsw.service = service
+
+	if lsw.store == nil {
+		return
+	}
 	lsw.perCluster = lsw.store.PerCluster(lsw.service, "")
 }
 
+func (lsw *loadStoreWrapper) updateLoadStore(store *load.Store) {
+	lsw.mu.Lock()
+	defer lsw.mu.Unlock()
+	if store == lsw.store {
+		return
+	}
+	lsw.store = store
+	lsw.perCluster = nil
+	if lsw.store != nil {
+		lsw.perCluster = lsw.store.PerCluster(lsw.service, "")
+	}
+
+}
+
 func (lsw *loadStoreWrapper) CallStarted(locality string) {
 	lsw.mu.RLock()
 	defer lsw.mu.RUnlock()
+	if lsw.perCluster == nil {
```

Comment:
The `perClusterStore` type in package `load` already checks for `nil`. Why is this required?

Reply:
Right, I forgot.
Deleted.